In [1]:
import pandas as pd
import requests

airlines_url = "https://en.wikipedia.org/wiki/List_of_airline_codes"

headers = {"User-Agent": "airline-codes-scraper/0.1 (your@email.com)"}
resp = requests.get(airlines_url, headers=headers, timeout=30)
resp.raise_for_status()

In [2]:
from io import StringIO
import pandas as pd

tables = pd.read_html(StringIO(resp.text))
print(len(tables))

1


In [3]:
df_airlines = tables[0]

In [4]:
df_airlines.shape

(6358, 6)

In [5]:
df_airlines.info()

<class 'pandas.DataFrame'>
RangeIndex: 6358 entries, 0 to 6357
Data columns (total 6 columns):
 #   Column                Non-Null Count  Dtype
---  ------                --------------  -----
 0   IATA                  1565 non-null   str  
 1   ICAO                  6243 non-null   str  
 2   Airline               6357 non-null   str  
 3   Telephony Designator  5855 non-null   str  
 4   Country/Region        6345 non-null   str  
 5   Comments              1018 non-null   str  
dtypes: str(6)
memory usage: 298.2 KB


In [56]:
for column in df_airlines:
    n = (df_airlines[column].isna()).sum()
    if n > 0:
        print(f"{column}: {n}")

IATA: 4793
ICAO: 115
Airline: 1
Telephony Designator: 503
Country/Region: 13
Comments: 5340


In [13]:
df_airlines[df_airlines["Airline"].isna()]

,IATA,ICAO,Airline,Telephony Designator,Country/Region,Comments
3428,NaN,JTN,NaN,JET TEST,United States,NaN


In [55]:
df_airlines["Country/Region"].isna().sum()

np.int64(13)

In [15]:
df_airlines[df_airlines["Country/Region"].isna()]

,IATA,ICAO,Airline,Telephony Designator,Country/Region,Comments
1,NaN,EVY,"34 Squadron, Royal Australian Air Force",Multiple,NaN,NaN
1572,NaN,NaN,BAX Global,NaN,NaN,NaN
3066,GY,NaN,Guyana Airways 2000,NaN,NaN,NaN
3250,NaN,IAC,INTERCHARTER,Romania,NaN,NaN
3278,NaN,YYY,ICAO,NaN,NaN,Used by airlines without a specific code
4125,NaN,MFT,Multiflight,YORKAIR,NaN,NaN
4682,8Q,NaN,Princess Air,NaN,NaN,no longer assigned
4796,REB,Rebus,REBUS,Bulgaria,NaN,NaN
5142,8S,NaN,Scorpio Aviation,NaN,NaN,NaN
5175,NaN,SPX,Spark Express (Glow),Glow|United States,NaN,NaN


In [31]:
for column in df_airlines:
    n = (df_airlines[column] == "n/a").sum()
    if n > 0:
        print(f"{column}: {n}")

In [32]:
df_airlines[df_airlines["Airline"] == "Travelsky"] 

,IATA,ICAO,Airline,Telephony Designator,Country/Region,Comments
5,1E,NaN,Travelsky,NaN,China,Local distribution system and hosting system (...


In [38]:
df_airlines[df_airlines["IATA"].isna() & df_airlines["ICAO"].isna()].head(2)

,IATA,ICAO,Airline,Telephony Designator,Country/Region,Comments
59,NaN,NaN,Ace Air,ACE TAXI,South Korea,defunct as of September 2015
245,NaN,NaN,Aero Comondu,AERO COMONDU,Mexico,NaN


In [39]:
(df_airlines["IATA"].isna() & df_airlines["ICAO"].isna()).sum()

np.int64(26)

In [54]:
df_airlines[df_airlines.duplicated(keep=False)]

,IATA,ICAO,Airline,Telephony Designator,Country/Region,Comments
1163,JS,KOR,Air Koryo,AIR KORYO,North Korea,NaN
1747,NaN,BYF,San Carlos Flight Center,BAY FLIGHT,United States,NaN
1866,NaN,IRO,CSA Air,IRON AIR,United States,NaN
2224,NaN,IRO,CSA Air,IRON AIR,United States,NaN
2245,PN,CHB,West Air (China),WEST CHINA,China,NaN
3688,JS,KOR,Air Koryo,AIR KORYO,North Korea,NaN
4789,NaN,RCB,Real Aero Club De Baleares,BALEARES,Spain,NaN
4795,NaN,RCB,Real Aero Club De Baleares,BALEARES,Spain,NaN
4974,NaN,BYF,San Carlos Flight Center,BAY FLIGHT,United States,NaN
6196,PN,CHB,West Air (China),WEST CHINA,China,NaN


## EDA discoveries

- Alot of missing values
- More ICAO codes than IATA codes
- 6 duplicates (12 rows)
- 26 rows both missing ICAO and IATA codes
- 13 rows are missing Country/Region
- 1 row eith missing Airline name (have IACO and Telephony Designator is JET TEST)

amount null values:
IATA: 4793   
ICAO: 115   
Airline: 1    
Telephony Designator: 503   
Country/Region: 13  
Comments: 5340  